app/scripts/directives/routeServicePie.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+angular.module('openshiftConsole')
+  .directive('routeServicePie', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        route: '=',
+      },
+      template: '<div ng-attr-id="{{chartId}}"></div>',
+      link: function($scope) {
+        var chart, config;
+
+        $scope.chartId = _.uniqueId('route-service-chart-');
+
+        config = {
+          bindto: '#' + $scope.chartId,
+          color: {
+            pattern: [$.pfPaletteColors.blue, $.pfPaletteColors.orange, $.pfPaletteColors.green, $.pfPaletteColors.red]
```

Comment:
what does c3 do if it needs more colors, start adjusting the shade of the color?